bpf/bpf_lxc.c

```diff
@@ -554,6 +554,22 @@ static inline int handle_ipv4_from_lxc(struct __sk_buff *skb, __u32 *dstID)
 			ep_tail_call(skb, CILIUM_CALL_IPV4_NODEPORT_REVNAT);
 			return DROP_MISSED_TAIL_CALL;
 		}
+# ifdef ENABLE_DSR
+		if (ct_state.dsr) {
+			struct ipv4_ct_tuple nat_tup = tuple;
+			nat_tup.flags = NAT_DIR_EGRESS;
+			nat_tup.sport = tuple.dport;
+			nat_tup.dport = tuple.sport;
+			struct ipv4_nat_entry *entry = snat_v4_lookup(&nat_tup);
+			if (entry) {
+				int ret1 = snat_v4_rewrite_egress(skb, &nat_tup,
+								  entry, l4_off);
```

Comment:
nit: could we just reuse `ret`? Also, I'd stick with current style and move declaration of `struct ipv4_nat_entry *entry` up, then leave a newline before the nat_tup overrides.